packages/fxa-auth-server/lib/payments/capability.ts

```diff
@@ -76,16 +75,44 @@ export class CapabilityService {
     );
   }
 
+  /**
+   * Create a price id changeset, by returning a priorPriceIds that is a disjoint
+   * set from the currentPriceIds given the affectedPriceIds.
+   *
+   * Due to the asynchronous nature of Stripe, Google, and Apple, we have no method
+   * that lets us know with certainty what price ids the user had active before the
+   * incoming event that triggered this change. To compensate for this, we assume
+   * that whatever the new current state of a users price ids are, that they were
+   * different than before this event. This does imply that we may broadcast that a
+   * user has had a capability removed or added multiple times even if they already
+   * had it, but relying parties can handle this gracefully.
+   */
+  private createPriceIdChangeset({
+    currentPriceIds,
+    affectedPriceIds,
+  }: {
+    currentPriceIds: string[];
+    affectedPriceIds: string[];
+  }) {
+    const priorPriceIds = new Set([...currentPriceIds, ...affectedPriceIds]);
+    for (const affectedPriceId of affectedPriceIds) {
+      if (currentPriceIds.includes(affectedPriceId)) {
+        // Remove the price id from the prior list for processing to assume that it
+        // was previously inactive and ensure we broadcast a change.
+        priorPriceIds.delete(affectedPriceId);
+      }
+    }
+    return [...priorPriceIds];
+  }
```

Comment:
Now that this is in one function, we can add a test for this logic.  I think this is tested indirectly right now?

Reply:
That's correct, its tested indirectly at the moment.